kratos/future/solving_strategies/schemes/implicit_scheme.h

```diff
@@ -0,0 +1,1772 @@
+        if (build_settings.Has("name")) {
+            const std::string builder_type = build_settings["name"].GetString();
+            if (builder_type == "block_builder") {
+                mpBuilder = Kratos::make_unique<BlockBuilderType>(rModelPart, build_settings); // TODO: Use the registry in here
+            } else if (builder_type == "elimination_builder") {
+                mpBuilder = Kratos::make_unique<EliminationBuilderType>(rModelPart, build_settings); // TODO: Use the registry in here
+            } else {
+                KRATOS_ERROR << "Wrong builder type \'" << builder_type << "\'. Available options are \'block_builder\' and \'elimination_builder\'." << std::endl;
+            }
+        } else {
+            KRATOS_WARNING("ImplicitScheme") << "Builder type not provided. Defaulting to \'block_builder\'." << std::endl;
+            mpBuilder = Kratos::make_unique<BlockBuilderType>(rModelPart, build_settings); // TODO: Use the registry in here
+        }
+    }
+
+    /// @brief Copy constructor
+    /// @param rOther Other ImplicitScheme
+    explicit ImplicitScheme(ImplicitScheme& rOther)
+      : mSchemeIsInitialized(rOther.mSchemeIsInitialized)
+      , mSchemeSolutionStepIsInitialized(rOther.mSchemeSolutionStepIsInitialized)
+    {
+        //TODO: Check this... particularly the mpBuilder pointer
+    }
+
+    /// @brief Destructor
+    virtual ~ImplicitScheme() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Create method
+     * @param rModelPart Reference to the model part
+     * @param ThisParameters The configuration parameters
+     */
+    virtual typename ImplicitScheme<TSparseMatrixType, TSystemVectorType, TSparseGraphType>::Pointer Create(
+        ModelPart& rModelPart,
+        Parameters ThisParameters) const
+    {
+        return Kratos::make_shared<ImplicitScheme<TSparseMatrixType, TSystemVectorType, TSparseGraphType>>(rModelPart, ThisParameters);
+    }
+
+    /**
+     * @brief Clone method
+     * @return The pointer of the cloned ImplicitScheme
+     */
+    virtual typename ImplicitScheme<TSparseMatrixType, TSystemVectorType, TSparseGraphType>::Pointer Clone()
+    {
+        return Kratos::make_shared<ImplicitScheme<TSparseMatrixType, TSystemVectorType, TSparseGraphType>>(*this) ;
+    }
+
+    /**
+     * @brief This is the place to initialize the ImplicitScheme.
+     * @details This is intended to be called just once when the strategy is initialized
+     */
+    virtual void Initialize()
+    {
+        KRATOS_TRY
+
+        // Check if the Initialize has been already performed
+        if (!mSchemeIsInitialized) {
+            // Initialize elements, conditions and constraints
+            EntitiesUtilities::InitializeAllEntities(*mpModelPart);
+
+            // Set the flag to avoid calling this twice
+            mSchemeIsInitialized = true; //TODO: Discuss with the KTC if these should remain or not
```

Comment:
@KratosMultiphysics/technical-committee we said yes to remove all of these as this should be handled properly in the `AnalysisStage`.

Reply:
Done.